agent/src/main/java/com/teamscale/jacoco/agent/configuration/ProcessInformationRetriever.java

```diff
@@ -0,0 +1,66 @@
+package com.teamscale.jacoco.agent.configuration;
+
+import com.teamscale.client.ProcessInformation;
+import com.teamscale.report.util.ILogger;
+
+import java.lang.management.ManagementFactory;
+import java.lang.reflect.InvocationTargetException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * The ProcessInformationRetriever class is responsible for retrieving process information such as the host name and
+ * process ID.
+ */
+public class ProcessInformationRetriever {
+
+	private final ILogger logger;
+
+	public ProcessInformationRetriever(ILogger logger) {
+		this.logger = logger;
+	}
+
+	/**
+	 * Retrieves the process information, including the host name and process ID.
+	 */
+	public ProcessInformation getProcessInformation() {
+		String hostName = getHostName();
+		String processId = getPID();
+		return new ProcessInformation(hostName, processId, System.currentTimeMillis());
+	}
+
+	/**
+	 * Retrieves the host name of the local machine.
+	 */
+	private String getHostName() {
+		try {
+			InetAddress inetAddress = InetAddress.getLocalHost();
+			return inetAddress.getHostName();
+		} catch (UnknownHostException e) {
+			logger.error("Failed to determine hostname!", e);
+			return "";
```

Comment:
What about alternatives such as the IP address itself?

Reply:
AFAICS you'd retrieve the IP in Java via 
```
InetAddress ip = InetAddress.getLocalHost();
System.out.println("Your IP address is: " + ip.getHostAddress());
```
which is exactly the same function that we are currently catching the exception of. Do you see any alternative here?

Reply:
I would have expected that if you already have an ip address at hand (the `ip`) object, you could provided this information. I was assuming that the call to `getHostHame` could lead to an inverse lookup of the hostname as described in the docs ` reverse name lookup will be performed and the result will be returned based on the system configured name lookup service` in would have an exception in case this lookup fails. But I do not know the very details of the implementation here. See also https://stackoverflow.com/questions/7348711/recommended-way-to-get-hostname-in-java for fallback solutions.

Reply:
Which fallback solution are you referring to?

Reply:
I was referring to the alternatives to get the host machines name. But we can wait if this problem manifests in the end and fix it on demand. So this is fine for me for now.